### DISCUSSION TITLE
🏆  Admin-app 2024 golden path [DRAFT]

### DISCUSSION THREAD

**Discussion by matheusps** [vtex/shoreline#1289] [Technical RFCs]
[Thoughtworks Tech Radar](https://www.thoughtworks.com/en-br/radar)

## Frontend framework

- 🏆 Adopt: `raccoon`, `next-js (pages)`, `react > 17`.
- 🟢 Trial: `nextjs (app)`.
- 🟠 Assess: `remix`, `vite`.
- 🔴 Hold: `render-runtime`, `react < 17`, `gatsby`.

### TL;DR

- Use Shoreline Tokens.
- Prefer styling with Vanilla CSS, CSS-Modules, and Sass. 
- Prefer PostCSS as processor. 
- While using Sass, prefer the `.scss` syntax.
- Avoid atomic CSS and runtime CSS-In-JS. 

### The RTG

- 🏆 Adopt: `@vtex/shoreline-tokens`, `CSS`, `postcss`, `css-modules`, `sass`.
- 🟢 Trial: `lightningcss`.
- 🟠 Assess: `vanilla-extract`, `pandacss`.
- 🔴 Hold: `emotion`, `styled-components`, `stitches`, `tailwindcss`, `tachyons`.

## Testing [TBD]

- 🏆 Adopt: `jest`, `cypress`, `storybook`.
- 🟢 Trial: `playwright`, `chromatic`.
- 🟠 Assess: `vitest`.
- 🔴 Hold: TBD

1. Unit: Jest vs Vitest
2. E2E: Playwright vs Cypress

## Repository

- 🏆 Adopt: `turborepo`, `lerna`, `pnpm`, `husky`, `plopjs`, `commitlint`, `commitizen`, `eslint`, `prettier`.
- 🟢 Trial: `changesets`, `yarn`.
- 🟠 Assess: `nx`.
- 🔴 Hold: `yarn v2+`, `bun`.

## Localization

- 🏆 Adopt: `react-intl`, `crowdin`.
- 🟢 Trial: -.
- 🟠 Assess: `i18n-next`.
- 🔴 Hold: -.

## Host [TBD]

<img src="https://github.com/vtex/shoreline/assets/6964311/3e225725-f0ff-40aa-8950-ccf5b539fc9b" alt="here we go again" width="148px" height="148px" />

- 🏆 Adopt:  `Vercel`.
- 🟢 Trial: -.
- 🟠 Assess: -.
- 🔴 Hold: `Netlify`, `AWS-Amplify`, `Web-Ops`.

**Comment by marcelovicentegc** [comment]
### Frontend framework

If we are to place NextJS app router on trial, I think it's fair to move Remix there as well.

### Styling

BACK WITH THE OLD SCHOOLERS: LGTM 👍 

### Testing [TBD]

LGTM 👍 

### Repository

I only miss [semantic-release](https://github.com/semantic-release/semantic-release) in the list. I would encourage teams to adopt it as well. Personally, it's my favorite for its simplicity.

### Localization

It occurs to me that we could move `i18n-next` under our Adopt matrix. It's a stable framework and has been around for more time then react-intl (8 years +).

### Host

I know no one mentioned it, but it occurs to me that we should at least list Cloudflare Workers/Pages under hold. Also, I will go after some verdict after [NextJS Admin Apps Provider Proposal](https://docs.google.com/document/d/1-JwkUjvufNpc_g-EKk_KeM1Z_V9FFELrDhnrqxjKOMQ/edit#heading=h.ta72161di1y0).

### Documetation

We should encourage teams to use mermaid to document their systems.

**Comment by kevinch** [comment]
About Business Metrics (Analytics), we got a [PR being reviewed](https://github.com/vtex/internal-docs/pull/3116) for the internal docs.
